spec/cli_spec.rb

```diff
@@ -528,6 +528,53 @@
     end
   end
 
+  context "#set_replication" do
+    let(:replication_double) { double(:activate => true) }
```

Comment:
Rather than a double here, it would be better to use an actual instance of the class.

That way, rather than stubbing all the assignments, you will be able to assert on the actual value of the attributes. Then you just need to stub the `#activate` method.